Js storage component

aceDavon opened the pull request:
Create logic to persist verified form data to user's browser on submit

aceDavon committed:
Fix issues with variables on auth.js

SurafelGetachewT committed:
Fix variable errors

Lucas-Masaba commented on js/auth.js:
```diff
@@ -0,0 +1,69 @@
+const persistData = (status, data) => {
+  status = true;
+  localStorage.setItem('portfolioData', JSON.stringify(data));
+
```
- [ ] To get the data you stored on local storage and display it, use `localstorage.getItem(portfolioData)`  and render it. 👍

Lucas-Masaba reviewed (changes requested):
# Hello @aceDavon and @SurafelGetachewT  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

![almost-there](https://media.giphy.com/media/ReyD4OdRiHeLDCRaEP/giphy.gif)

## Highlights 🍾 
- Data is successfully being stored with localstorage 🏄 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

- [ ]  In general, your contact form is not working as expected.  When I type in data and refresh the page, the data is not stored on the local storage and is not persisted in the input fields. It's only when I click the submit button that data is stored in the local storage. When I click the submit button an error occurs but the data is stored on the local storage none the less.

aceDavon committed:
Create verification for form inputs

Extend verfification to other form inputs to correct issue raised from
last review.

aceDavon committed:
Rebase on new branch

aceDavon committed:
Merge branch 'main' into js-storage-component

Resolve conflicts from previous commits on the new branch

sumairq reviewed (approved):
### Hi Team , 

<img src="https://media4.giphy.com/media/BPJmthQ3YRwD6QqcVD/giphy.gif"/>

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Data is preserved in browser
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Github flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
